File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class WeeklyBugTrend(BaseModel):
    week_start: str
    bug_count: int
    security_count: int
    style_count: int
    performance_count: int


class WeeklyCost(BaseModel):
    week_start: str
    cost_usd: float
    prompt_tokens: int
    completion_tokens: int


class LatencyBucket(BaseModel):
    label: str
    count: int


class SeverityBreakdown(BaseModel):
    low: int
    medium: int
    high: int
    critical: int


class AnalyticsResponse(BaseModel):
    total_prs_reviewed: int
    avg_risk_score: float | None
    avg_latency_ms: float | None
    total_cost_usd: float
    severity_breakdown: SeverityBreakdown
    weekly_bug_trend: list[WeeklyBugTrend]
    weekly_cost: list[WeeklyCost]
    latency_buckets: list[LatencyBucket]

File: frontend/src/types/analytics.ts
export interface WeeklyBugTrend {
  week_start: string
  bug_count: number
  security_count: number
  style_count: number
  performance_count: number
}

export interface WeeklyCost {
  week_start: string
  cost_usd: number
  prompt_tokens: number
  completion_tokens: number
}

export interface LatencyBucket {
  label: string
  count: number
}

export interface SeverityBreakdown {
  low: number
  medium: number
  high: number
  critical: number
}

export interface Analytics {
  total_prs_reviewed: number
  avg_risk_score: number | null
  avg_latency_ms: number | null
  total_cost_usd: number
  severity_breakdown: SeverityBreakdown
  weekly_bug_trend: WeeklyBugTrend[]
  weekly_cost: WeeklyCost[]
  latency_buckets: LatencyBucket[]
}
